tutorials/networking/websocket.rst

```diff
@@ -99,63 +88,75 @@ This example will show you how to create a WebSocket server that listens for rem
-        print("Client %d connected with protocol: %s" % [id, proto])
-
-    func _close_request(id, code, reason):
-        # This is called when a client notifies that it wishes to close the connection,
-        # providing a reason string and close code.
-        print("Client %d disconnecting with code: %d, reason: %s" % [id, code, reason])
-
-    func _disconnected(id, was_clean = false):
-        # This is called when a client disconnects, "id" will be the one of the
-        # disconnecting client, "was_clean" will tell you if the disconnection
-        # was correctly notified by the remote peer before closing the socket.
-        print("Client %d disconnected, clean: %s" % [id, str(was_clean)])
-
-    func _on_data(id):
-        # Print the received packet, you MUST always use get_peer(id).get_packet to receive data,
-        # and not get_packet directly when not using the MultiplayerAPI.
-        var pkt = _server.get_peer(id).get_packet()
-        print("Got data from client %d: %s ... echoing" % [id, pkt.get_string_from_utf8()])
-        _server.get_peer(id).put_packet(pkt)
-
-    func _process(delta):
-        # Call this in _process or _physics_process.
-        # Data transfer, and signals emission will only happen when calling this function.
-        _server.poll()
+	extends Node
+
+	# The port we will listen to
+	const PORT = 9080
+
+	# Our WebSocketServer instance
+	var _server = WebSocketMultiplayerPeer.new()
+
+	# Keeps track of the connected peer IDs
+	var peer_ids: Array[int] = []
+
+
+	func _ready():
+		# Connect base signals to get notified of new client connections,
+		# and disconnections.
+		_server.peer_connected.connect(_connected)
+		_server.peer_disconnected.connect(_disconnected)
+
+		# Start listening on the given port.
+		var err = _server.create_server(PORT)
+		if err != OK:
+			print("Unable to start server")
+			set_process(false)
+
+
+	func _connected(id):
+		peer_ids.append(id)
+
+		# This is called when a new peer connects, "id" will be the assigned peer id,
+		print("Peer %d connected" % id)
+
+
+	func _disconnected(id):
+		peer_ids.erase(id)
+
+		# This is called when a client disconnects, "id" will be the one of the
+		# disconnecting client
+		var peer = _server.get_peer(id)
+		var code = peer.get_close_code()
+		var reason = peer.get_close_reason()
+		print("Peer %s closed with code: %d, reason %s. Clean: %s" % [id, code, reason, code != -1])
+
+
+	func _process(delta):
+		# Call this in _process or _physics_process.
+		# Data transfer, and signals emission will only happen when calling this function.
+		_server.poll()
+
+		for peer_id in peer_ids:
+			var peer = _server.get_peer(peer_id)
+			peer.poll()
+
+			var peer_state = peer.get_ready_state()
+			if peer_state == WebSocketPeer.STATE_OPEN:
+				while peer.get_available_packet_count():
+					var packet_text = peer.get_packet().get_string_from_utf8()
+					print(" Got data from peer %s: %s" % [peer_id, packet_text])
```

Comment:
```suggestion
					print("Got data from peer %d: %s ... echoing" % [peer_id, packet_text])
```
Alternatively change the printed message below, as this has changed, also `peer_id` is an integer, not a string here.